packages/adblocker-electron/src/index.ts

```diff
@@ -164,101 +158,65 @@ export class ElectronBlocker extends FiltersEngine {
+    url: string,
+    msg?: IBackgroundCallback,
+  ): Promise<{ active: boolean; error?: string }> => {
     const parsed = parse(url);
     const hostname = parsed.hostname || '';
     const domain = parsed.domain || '';
 
+    // `msg` is undefined for the initial call and present for subsequent updates
     const { active, styles, scripts, extended } = this.getCosmeticsFilters({
       domain,
       hostname,
       url,
 
-      // This needs to be done only once per frame
-      getBaseRules: true,
-      getInjectionRules: true,
-      getExtendedRules: true,
-      getRulesFromHostname: true,
-      getRulesFromDOM: false, // Only done on updates (see `onGetCosmeticFiltersUpdated`)
+      // DOM information, only available for updates
+      classes: msg?.classes,
+      hrefs: msg?.hrefs,
+      ids: msg?.ids,
+
+      // Rules to fetch: true for initial call, false for updates
+      getBaseRules: !msg,
+      getInjectionRules: !msg,
+      getExtendedRules: !msg,
+      getRulesFromHostname: !msg,
+
+      // Only true for update calls when we have DOM information
+      getRulesFromDOM: !!msg,
 
       callerContext: {
         frameId: event.frameId,
         processId: event.processId,
+        lifecycle: msg?.lifecycle,
       },
     });
 
     if (active === false) {
-      event.returnValue = null;
-      return;
+      return { active: false };
     }
 
-    // Inject custom stylesheets
-    this.injectStyles(event.sender, styles);
-
-    event.sender.send('get-cosmetic-filters-response', {
-      active,
-      extended,
-      styles: '',
-    } as IMessageFromBackground);
-
-    // to execute Inject scripts synchronously, simply return scripts to renderer.
-    event.returnValue = scripts;
-  };
-
-  public onGetCosmeticFiltersUpdated = (
-    event: Electron.IpcMainEvent,
-    url: string,
-    msg: IBackgroundCallback,
-  ): void => {
-    // Extract hostname from sender's URL
-    const parsed = parse(url);
-    const hostname = parsed.hostname || '';
-    const domain = parsed.domain || '';
-
-    const { active, styles, extended } = this.getCosmeticsFilters({
-      domain,
-      hostname,
-      url,
-
-      classes: msg.classes,
-      hrefs: msg.hrefs,
-      ids: msg.ids,
-
-      // Only done on first load in the frame, disable for updates
-      getBaseRules: false,
-      getInjectionRules: false,
-      getExtendedRules: false,
-      getRulesFromHostname: false,
-
-      // This will be done every time we get information about DOM mutation
-      getRulesFromDOM: true,
+    try {
+      if (styles.length > 0) {
+        await event.sender.insertCSS(styles, { cssOrigin: 'user' });
+      }
 
-      callerContext: {
-        frameId: event.frameId,
-        processId: event.processId,
+      for (const script of scripts) {
+        await event.sender.executeJavaScript(script, true);
+      }
 
-        lifecycle: msg.lifecycle,
-      },
-    });
+      if (extended.length > 0) { }
```

Comment:
From the original code (before this PR), I found that we don't support extended CSS for electron. Let's disable it from match call for now.